mkosi/config.py

```diff
@@ -4062,19 +4061,19 @@ def parse_new_includes(self) -> None:
             with chdir(path if path.is_dir() else Path.cwd()):
                 self.parse_config_one(path if path.is_file() else Path("."))
 
-    def finalize_value(self, setting: ConfigSetting) -> Optional[Any]:
+    def finalize_value(self, setting: ConfigSetting[T]) -> Optional[T]:
         # If a value was specified on the CLI, it always takes priority. If the setting is a collection of
         # values, we merge the value from the CLI with the value from the configuration, making sure that the
         # value from the CLI always takes priority.
         if hasattr(self.cli, setting.dest) and (v := getattr(self.cli, setting.dest)) is not None:
             if isinstance(v, list):
-                return (getattr(self.config, setting.dest, None) or []) + v
+                return (getattr(self.config, setting.dest, None) or []) + v  # type: ignore
             elif isinstance(v, dict):
-                return (getattr(self.config, setting.dest, None) or {}) | v
+                return (getattr(self.config, setting.dest, None) or {}) | v  # type: ignore
             elif isinstance(v, set):
-                return (getattr(self.config, setting.dest, None) or set()) | v
+                return (getattr(self.config, setting.dest, None) or set()) | v  # type: ignore
             else:
-                return v
+                return v  # type: ignore
```

Comment:
I think this is a place where we can do explicit casts instead, since we can guarantee more than just ignoring this. Same in the next hunk.

Reply:
I tried that but problem is we would need to cast to `list[something]` but we cannot extract that "something" from T (and casting to T also does not work). I could take another look but I played around with this a bit to no avail

Reply:
I got it a bit more strict but currently type checkers are still not really able to follow the code / Pythons type system is not capable enough to express what we are doing here. It _could_ be possible to do some trickery with `TypeVar`s, several `@overloads`, and putting the collection merging into helper functions but I failed to make that work in an acceptable amount of time so I will let this rest for now